File: projects/picker/apps/cross-chain-pay-one-usdt/entry.js
const fs = require('fs');
const path = require('path');
const toml = require('@iarna/toml');
const { main } = require('./src/main');

function sleep(ms) {
    return new Promise(resolve => setTimeout(resolve, ms));
}

async function entry() {
    // 获取 config.toml 在主机上的绝对路径，兼容Windows和Unix系统
    const configPath = path.join(path.dirname(__filename), 'config.toml');
    let config;
    
    try {
        // 读取 config.toml
        const configContent = fs.readFileSync(configPath, 'utf8');
        config = toml.parse(configContent);
        
        try {
            // 确保 task 对象存在
            if (!config.task) {
                config.task = {};
            }
            
            // 设置任务状态为 Running
            config.task.status = 'Running';
            fs.writeFileSync(configPath, toml.stringify(config));
            
            // 调用主函数执行跨链转账
            await main();
            
            // 等待1秒
            await sleep(1000);
            
        } catch (error) {
            // 更新任务状态为 Error
            config.task.status = 'Error';
            fs.writeFileSync(configPath, toml.stringify(config));
            throw error;
        }
        
        // 更新任务状态为 Idle
        config.task.status = 'Idle';
        fs.writeFileSync(configPath, toml.stringify(config));
        
    } catch (error) {
        if (error.code === 'ENOENT') {
            console.error('Error: config.toml not found.');
        } else if (error.name === 'TomlDecodeError') {
            console.error(`Error: config.toml is malformed: ${error.message}`);
        } else {
            throw error;
        }
    }
}

if (require.main === module) {
    entry();
}

module.exports = { entry };

File: projects/picker/apps/picker-python-project/entry.py
import os
import toml
from src.main import main
from time import sleep

def entry():
    # 获取 config.toml 在主机上的绝对路径，兼容Windows和Unix系统
    config_path = os.path.join(os.path.dirname(__file__), "config.toml")

    # 读取 config.toml
    config = toml.load(config_path)

    try:
        config["task"]["status"] = "Running"
        with open(config_path, "w") as f:
            toml.dump(config, f)
        main()
        sleep(1)
    except FileNotFoundError:
        print("Error: config.toml not found.")
        return
    except toml.TomlDecodeError as e:
        print(f"Error: config.toml is malformed: {e}")
        return
    except Exception:
        config["task"]["status"] = "Error"
        with open(config_path, "w") as f:
            toml.dump(config, f)
        raise

    config["task"]["status"] = "Idle"
    with open(config_path, "w") as f:
        toml.dump(config, f)

if __name__ == "__main__":

    entry()
